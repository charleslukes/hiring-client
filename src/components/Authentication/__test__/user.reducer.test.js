import user, { initialState } from '../user.reducer';

describe('INITIAL_STATE', () => {
  test('is correct', () => {
    const action = { type: 'dummy_action' };
    const initialState = { loading: false, error: '', data: {} };

    expect(user(undefined, action)).toEqual(initialState);
  });
});

describe('USER_LOGIN', () => {
  test('returns the correct state', () => {
    const action = {
      type: 'USER_LOGIN',
      payload: {
        dummy: 'user data'
      }
    };
    const expectedState = {
      ...initialState,
      data: {
        dummy: 'user data'
      }
    };

    expect(user(initialState, action)).toEqual(expectedState);
  });
});

describe('LOADING', () => {
  test('returns the correct state', () => {
    const action = {
      type: 'LOADING',
      payload: true
    };
    const expectedState = {
      ...initialState,
      loading: true
    };

    expect(user(initialState, action)).toEqual(expectedState);
  });
});

describe('USER_LOGIN_ERROR', () => {
  test('returns the correct state', () => {
    const action = {
      type: 'USER_LOGIN_ERROR',
      payload: 'error message'
    };
    const expectedState = {
      ...initialState,
      error: 'error message'
    };

    expect(user(initialState, action)).toEqual(expectedState);
  });
});
