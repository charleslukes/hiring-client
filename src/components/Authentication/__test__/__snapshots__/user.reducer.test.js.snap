// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INITIAL_STATE is correct 1`] = `
Object {
  "data": Object {},
  "error": "",
  "loading": false,
}
`;

exports[`SELECT_AVATAR returns the correct state 1`] = `
Object {
  "data": Object {
    "dummy": "user data",
  },
  "error": "",
  "loading": false,
}
`;

exports[`USER_LOGIN returns the correct state 1`] = `
Object {
  "data": Object {
    "dummy": "user data",
  },
  "error": "",
  "loading": false,
}
`;
